### PR TITLE
AP_Scheduler: allow for loop rates above 400Hz

### PR DESCRIPTION
**Pull request by tridge**
with ChibiOS we can run copter at 1KHz with no scheduling misses. This
small PR allows that to be configured.

### CODE REVIEW

**Comment by magicrub** [comment]
I'm curious. If no data has been updated could run this at 4000 Hz? I'm assuming if all the data is stale then not much happens. However, if a sample from, lets say the baro, comes in between the IMU samples then the whole EKF loops refreshes it's states so in theory it actually would be running fast with all the various async data updates. Hmm, or is the 1000 Hz scheduler loop a limiting factor where no driver would update data faster than 1000 Hz? ALTHOUGH if data is coming in on a different thread such as UAVCAN or Serial, then I suppose it's possible again?

Also, shouldn't we be wrapping this with a 400Hz limit on NUTTX but 1000 or 1500 on ChibiOS? What about Linux?

I'm curious!

**Comment by Pedals2Paddles** [comment]
Could the rate automatically throttle itself up and down based on scheduler performance? Of course with hard minimum and maximum limits.

**Comment by magicrub** [comment]
@Pedals2Paddles Hmm, I'd hate for it to change too often but I don't think there's any technical reason to not have that. Although reviewing EKF logs might get harder if the rate keeps changing

**Comment by tridge** [comment]
I'm waiting for the first flight tests of this from Leonard. I'll update the comment docs when we know how well it works.
Values above 1000 will need changes to AP_InertialSensor as we currently average the IMUs down to 1kHz even if sampling at 8kHz

File: libraries/AP_Scheduler/AP_Scheduler.cpp [hunk 1]
```diff
@@ -63,8 +63,8 @@ AP_Scheduler::AP_Scheduler(void)
     // only allow 50 to 400 Hz
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_Scheduler/AP_Scheduler.cpp]: fixed

**Comment by tridge** [comment]
flight tests went well, merging

**Pull request merged**